server/lib/UserManager.php

```diff
@@ -270,7 +241,85 @@ public function register(Request $request, Response $response) {
 		return $response;
 	}
 
-	/**
+    public function delete(Request $request, Response $response) {
+        $body = json_decode($request->getBody(), true);
+
+        if ($body === null || !isset($body['message']) || !isset($body['message']['data']) ||
+            !isset($body['message']['data']['federationId']) || !isset($body['signature']) ||
+            !isset($body['message']['timestamp'])) {
+            $response->withStatus(400);
+            return $response;
+        }
+
+        $cloudId = $body['message']['data']['federationId'];
+
+        try {
+            $verified = $this->verifyRequest($cloudId, $body['message'], $body['signature']);
+        }  catch(\Exception $e) {
+            $response->withStatus(400);
+            return $response;
+        }
+
+
+        if ($verified) {
+            $result = $this->deleteDBRecord($cloudId);
+            if ($result === false) {
+                $response->withStatus(403);
+            }
+        } else {
+            // ERROR OUT
+            $response->withStatus(403);
```

Comment:
well if we failed to verify the signature I think 403 is correct. That's also what we do  in other places when verifying the signature fails, so let's be consistent.